src/librustc_mir/interpret/eval_context.rs

```diff
@@ -768,7 +768,10 @@ impl<'mir, 'tcx, M: Machine<'mir, 'tcx>> InterpCx<'mir, 'tcx, M> {
         // FIXME(oli-obk): make this check an assertion that it's not a static here
         // FIXME(RalfJ, oli-obk): document that `Place::Static` can never be anything but a static
         // and `ConstValue::Unevaluated` can never be a static
-        let param_env = if self.tcx.is_static(gid.instance.def_id()) {
+        // FIXME(oli-obk, spastorino): the above FIXME is not true anymore, PlaceBase::Static does
+        // not exist anymore (except for promoteds but it's going away soon).
```

Comment:
This chain of FIXMEs is getting some weird kind of joke.^^

I think it referred to the "lazy" vs "evaluated" AllocId for statics?

Reply:
The last two are gone in my PR :smile: